### PR TITLE
atomic transaction for creating new user

### PR DESCRIPTION
**Pull request by mikedebock**
Atomic transaction for creating new user, when firing multiple requests at same time.

### CODE REVIEW

File: src/django_cognito_jwt/backend.py [hunk 1]
```diff
@@ -26,6 +27,7 @@ def __init__(self):
 
         super().__init__()
 
+    @transaction.atomic
```
Comment (mvantellingen) [inline src/django_cognito_jwt/backend.py:30]: We should only wrap line 54 in the atomic clause

File: src/django_cognito_jwt/backend.py [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import logging
 
+from django.db import transaction
```
Comment (mvantellingen) [inline src/django_cognito_jwt/backend.py:3]: ordering ;p

**Review by mvantellingen — CHANGES_REQUESTED** [review]

**Comment by mvantellingen** [comment]
BTW; using a transaction can still result in an error when concurrent requests are done. This should be handled with a try/except clause (select user, insert if no user, if insert fails due to duplicate, retrieve user)

**Comment by mikedebock** [comment]
This pr is unnecessary, fixed in get_or_create_for_cognito method.

**Pull request closed without merging**